# Bepaling van warmtecapaciteit van een onbekend materiaal


# Introductie

Onbekende materialen kunnen geïdentificeerd worden door hun eigenschappen te meten. Een van deze eigenschappen is de warmtecapaciteit. In dit practicum gaan we de warmtecapaciteit van een onbekend materiaal bepalen door middel van een calorimeter experiment. Daarbij wordt een bepaalde massa van het materiaal naar een bekende temperatuur gebracht waarna het in een bekende hoeveelheid water met bekende temperatuur wordt geplaatst. Door de temperatuur van het water te meten na het mengen kan de warmtecapaciteit van het onbekende materiaal worden berekend.

# Theorie

De soortelijke warmte $c$ van een materiaal is gedefinieerd als de hoeveelheid warmte $Q$ die nodig is om de temperatuur $T$ van een kilogram van het materiaal met één graad Celsius (of één Kelvin) te verhogen:

$$
    c = \frac{Q}{m \Delta T}
$$ (eq:heat_capacity)

Waarbij $Q$ de hoeveelheid warmte in Joules is, $m$ de massa in kilogram is en $\Delta T$ de verandering in temperatuur is. Gegeven de wet van Black, die stelt dat de totale hoeveelheid warmte in een geïsoleerd systeem constant blijft, kunnen we de warmte die het onbekende materiaal verliest gelijkstellen aan de warmte die het water opneemt:

$$
    Q_{materiaal} = -Q_{water} 
$$ (eq:black)

wanneer we de massa's en de begintemperaturen van beide systemen kennen, maar slechts een van de twee soortelijke warmtes, kunnen we de onbekende soortelijke warmte berekenen. We combineren vergelijkingen [](#eq:heat_capacity) en [](#eq:black) om de volgende vergelijking te krijgen:

$$
    T_e = \frac{c_w m_w T_{w,b}+c_m m_m T_{m,b}}{c_w m_w + c_m m_m}
$$ (eq:combined)

Waarbij de subscripts $b$ en $e$ respectievelijk staan voor begintoestand en eindtoestand, $w$ voor water en $m$ voor het onbekende materiaal.

Bij metingen aan verschillende massa's van het onbekende materiaal en vervolgens een least square fit aan bovenstaande vergelijking kunnen we een precieze waarde voor de soortelijke warmte van het onbekende materiaal bepalen. Dat is, wanneer de warmtecapaciteit van bijvoorbeeld de beker te verwaarlozen is.


# Methode en materialen

## Ontwerp
De bovenstaande theorie wordt gebruikt om de soortelijke warmte van een onbekend materiaal te bepalen. Het experiment bestaat uit het verwarmen van verschillende massa's van het onbekende materiaal tot een bekende temperatuur, waarna het in een bekende hoeveelheid water met bekende temperatuur wordt geplaats. Door de temperatuur van het water te meten na het mengen kan de warmtecapaciteit van het onbekende materiaal worden berekend. Om de tijd voor het meten van meerdere materialen te reduceren, worden de data van de verschillende groepen in het lokaal samengevoegd. Van tevoren is afgesproken welke massa's door welke groep worden gemeten, en hoeveel water er gebruikt wordt.

## Materialen
Hieronder staat de lijst van benodigde materialen bij deze proef:
- Calorimeter
- Thermometer of temperatuursensor
- Verwarmingsbron 
- Diverse massablokjes van onbekend materiaal
- Weegschaal
- Water
- Maatcilinder of maatbeker


```{figure} figures/c_onbekend_setup.jpg
:width: 70%
:label: fig_c_onbekend_setup

Een schematische weergave van de opstelling
```


## Procedure
Bespreek wie welke massa's van het onbekende materiaal gaat meten.
Bespreek ook hoeveel water er gebruikt gaat worden.
Bepaal de begintemperaturen.
Hevel het aantal afgesproken massa's in de maatbeker. 
Roer voorzicht zodat de temperatuur homogeen is.
Noteer de hoogste gemeten temperatuur, dit is $T_e$.
Wissel de metingen uit met de andere groepen en voer de data-analyse uit.

```{note}
Hieronder staat een voorbeeld hoe je een grafiek moet opslaan en vervolgens in je document kunt oproepen.
Het voorbeeld kun je natuurlijk straks verwijderen.
```

# Resultaten


In [ ]:
import numpy as np
import matplotlib.pyplot as plt
from scipy.optimize import curve_fit

T_water_blokje = 71.2 #graden Celcius
T_begin = np.array([21.5, 20.8, 23.1, 24.9]) #graden Celcius
T_eind = np.array([22.4, 23.1, 24.9, 26.0]) #graden Celcius
m_water = np.array([0.2649, 0.1307, 0.1307, 0.1307]) #kilogram
m_blokje = np.array([0.0518, 0.0653, 0.0654, 0.0389]) #kilogram

c_water = 4186 #J/kg*K

**Notes**

De metingen wijken af van elkaar, doordat de starttemperatuur telkens is gevarieerd. Ook is bij de eerste meting de massa van het water anders dan bij de volgende metingen. Dit heeft er voor gezorgd dat de uiteindelijke waarde die we hebben gekregen voor de warmtecapaciteit

In [14]:
dT_water = np.abs(T_eind - T_begin) #graden Kelvin
dT_blok = np.abs(T_eind - T_water_blokje) #graden Kelvin

c_blok_individueel = (c_water * m_water * dT_water) / (m_blokje * dT_blok) #J/kgK

mean_c_individueel = c_blok_individueel.mean() #J/kgK
std_c_individueel = c_blok_individueel.std(ddof=1) #J/kgK

print("Directe berekening per meting (c_m in J/kgK):")
for i, c_val in enumerate(c_blok_individueel, start=1):
    print(f"Meting {i}: c_m = {c_val:.1f} J/kgK")
print(f"\nGemiddelde warmtecapaciteit (individueel): {mean_c_individueel:.1f} J/kgK")
print(f"Standaardafwijking (individueel): {std_c_individueel:.1f} J/kgK")

# Function fit

T_mb = np.full_like(T_begin, T_water_blokje)
T_wb = T_begin

def T_e_model(m_block, c_m):
    a = c_water * m_water * T_wb + c_m * m_block * T_mb
    b = c_water * m_water + c_m * m_block
    return a / b

popt, pcov = curve_fit(T_e_model, m_blokje, T_eind, p0=[400.0])
c_fit = popt[0]
c_fit_err = np.sqrt(np.diag(pcov))[0]

print(f"De warmtecapaciteit = {c_fit:.1f} ± {c_fit_err:.1f} J/kgK (Functie fit)")

Directe berekening per meting (c_m in J/kgK):
Meting 1: c_m = 394.8 J/kgK
Meting 2: c_m = 400.6 J/kgK
Meting 3: c_m = 325.2 J/kgK
Meting 4: c_m = 342.3 J/kgK

Gemiddelde warmtecapaciteit (individueel): 365.7 J/kgK
Standaardafwijking (individueel): 37.7 J/kgK
De warmtecapaciteit = 363.6 ± 20.5 J/kgK (Functie fit)


# Conclusie

De metingen laten zien dat de warmtecapaciteit van de onbekende stof ongeveer 364 ± 20.5 J/kgK is. Dit komt redelijk overeen met de warmtecapaciteit van messing, koper en brons, die respectievelijk 377 J/kgK, 380 J/kgK en 380 J/kgK zijn. Als de onzekerheid wordt meegenomen kan dus geconcludeerd worden dat het onbekende materiaal waarschijnlijk een van deze drie metalen is.

# Discussie

De metingen wijken wel erg van elkaar af. Dit kan komen doordat er warmte verloren is gegaan aan de omgeving, of doordat de temperatuur niet goed gemeten is. Om de nauwkeurigheid van de metingen te verbeteren kan er bijvoorbeeld een beter geïsoleerde calorimeter worden gebruikt. Ook kan grotere temperatuurverschillen en herhaling van de metingen worden gebruikt om de invloed van meetfouten te verminderen.

